Added CasaTunes support

resoai opened the pull request:
## Description:

This component adds support for CasaTunes multiroom audio servers. Zones are added as media players which can select source, volume and be muted. Sources are added as media players that feed audio to the zones and can support play/pause/stop/backward/forward commands as well as album cover and track position.


## Example entry for `configuration.yaml`:
```yaml
media_player:
  - platform: casatunes
    name: CasaTunes
    host: 192.168.1.31

```

## Checklist:
  - [x] The code change is tested and works locally.
  - [x] Local tests pass with `tox`. 
  - [x] There is no commented out code in this PR.

If user exposed functionality or configuration variables are added/changed:
  - [ ] Documentation added/updated in [home-assistant.io](https://github.com/home-assistant/home-assistant.io)

If the code communicates with devices, web services, or third-party tools:
  - [x] New dependencies have been added to the `REQUIREMENTS`
  - [x] New dependencies are only imported inside functions that use them

If the code does not interact with devices:
  - [x] Tests have been added to verify that the new code works.

balloob committed:
Bumped version to 0.89.0b0

balloob committed:
Version bump to 0.90.0b0

robbiet480 committed:
Mobile App: Require encryption for registrations that support it (#21852)

## Description:

**Related issue (if applicable):** fixes #21758

## Checklist:
  - [x] The code change is tested and works locally.
  - [x] Local tests pass with `tox`. **Your PR cannot be merged unless tests pass**
  - [x] There is no commented out code in this PR.

robbiet480 committed:
Mobile App: Support rendering multiple templates at once (#21851)

* Support rendering multiple templates at once

* Only catch TemplateError and dont log the error

robbiet480 committed:
Mobile App: Update Location schema updates & device ID generation (#21849)

* Update location schema

* Generate a random device ID at registration time for later use with device_tracker.see

* Remove host name from device_tracker.see payload

* Drop consider_home from the payload

* Remove stale consider_home in schema

* Remove source_type

philhawthorne committed:
Remove UTF8 decoding for Waze (#22020)

Removes the UFT8 decoding for the Waze sensor, which broke in 0.89

Fixes #21739

robbiet480 committed:
Mobile App: Enable loading via discovery (surprise inside!) (#22027)

![](http://funpeep.com/wp-content/uploads/2014/04/Cute-White-Cat-Wallpaper.jpg)

robbiet480 committed:
Mobile App: Discovery to default configuration.yaml, zeroconf to default_config (#22028)

* Move discovery into default configuration.yaml

* Add zeroconf to default_config

JeffLIrion committed:
Bump androidtv to 0.0.11 (#22025)

mvn23 committed:
Bump pyotgw to 0.4b2 (#21973)

awarecan committed:
Fix lifx light async error (#22031)

cgtobi committed:
Bring back the boiler status (#22021)

emontnemery committed:
Update additional platforms to use new MQTT message callback (#22030)

* Move additional platforms to new MQTT callback

* Fix automation.mqtt

robbiet480 committed:
Mobile App: Register devices into the registry (#21856)

* Register devices into the registry

* Switch to device ID instead of webhook ID

* Rearchitect mobile_app to support config entries

* Kill DATA_REGISTRATIONS by migrating registrations into config entries

* Fix tests

* Improve how we get the config_entry_id

* Remove single_instance_allowed

* Simplify setup_registration

* Move webhook registering functions into __init__.py since they are only ever used once

* Kill get_registration websocket command

* Support description_placeholders in async_abort

* Add link to mobile_app implementing apps in abort dialog

* Store config entry and device registry entry in hass.data instead of looking it up

* Add testing to ensure that the config entry is created at registration

* Fix busted async_abort test

* Remove unnecessary check for entry is None

pvizeli committed:
Fix Google Assistant User with Cloud (#22042)

* Fix Google Assistant User with Cloud

* Fix User Agent ID

* respell

* Fix object

* Fix tests

* fix lint

* Fix lint

balloob committed:
Bumped version to 0.90.0b1

robbiet480 committed:
Mobile App: Sensors (#21854)

## Description:

**Related issue (if applicable):** fixes #21782

## Checklist:
  - [x] The code change is tested and works locally.
  - [x] Local tests pass with `tox`. **Your PR cannot be merged unless tests pass**
  - [x] There is no commented out code in this PR.

robbiet480 committed:
Mobile App: Expose Cloud Remote UI FQDN in registration response (#22055)

* Add a callback to get the cloud remote UI FQDN

* Expose Cloud Remote UI FQDN in the registration response

* Return a URL instead of FQDN

balloob committed:
Return config entry ID after creation (#22060)

balloob committed:
Fix test

balloob committed:
Fix more test

awarecan committed:
Correct context (#22061)

pvizeli committed:
Binary Sensor for Remote UI & Fix timezone (#22076)

* Binary Sensor for Remote UI

* Fix lint

* Revert make hass public

* Add tests

balloob committed:
Updated frontend to 20190315.0

balloob committed:
Bumped version to 0.90.0b2

balloob committed:
Fix func

JeffLIrion committed:
Bump androidtv to 0.0.12 (#22072)

balloob committed:
Prevent cloud remote UI when using 127.0.0.1 as trusted network (#22093)

* Prevent cloud remote UI when using trusted networks

* Limit to 127.0.0.1 trusted network

* Update error msg

* Disable ipv6 loopback

balloob committed:
Updated frontend to 20190315.1

hunterjm committed:
Beta Fix: FFMPEG and Stream component (#22091)

* remove stream_source from ffmpeg and onvif and add to generic ip cam

* fix tests

balloob committed:
Bumped version to 0.90.0b3

mvn23 committed:
Fix opentherm_gw blocks HA startup when gateway unreachable. (#22106)

mvn23 committed:
Fix TypeError in current_temperature if no temperature is known. (#22112)

Don't set opentherm_gw climate temperatures to 0 on init.

mvn23 committed:
Improve opentherm gw startup (#22121)

* Improve fix in c90f0d5 (#22106). Schedule connect coroutine directly on the loop rather than waiting for EVENT_HOMEASSISTANT_START.

* Remove unused import.

hunterjm committed:
delete previously removed service option from services yaml (#22123)

balloob committed:
Updated frontend to 20190316.0

balloob committed:
Bumped version to 0.90.0b4

awarecan committed:
Remove hass.config from aws_lambda notify payload (#22125)

balloob committed:
Allow non-admins to listen to certain events (#22137)

WebSpider committed:
Bump tado version (#22145)

* Bump python-tado, new API endpoint

* Change references of old API endpoint to new

* Update REQUIREMENTS

hunterjm committed:
Fix resetting access token on streams with keepalive (#22148)

balloob committed:
Updated frontend to 20190318.0

frenck committed:
Upgrade toonapilib to 3.2.2 + lower interval (#22160)

pvizeli committed:
Remove config check over supervisor (#22156)

* Remove config check over supervisor

* Fix lint

* Fix tests

balloob committed:
Bumped version to 0.90.0b5

uchagani committed:
bump total_connect_client to 0.24 (#22166)

pvizeli committed:
Update Hass-NabuCasa 0.8 (#22177)

balloob committed:
Always load Hass.io component on Hass.io (#22185)

* Always load Hass.io component on Hass.io

* Lint

* Lint

balloob committed:
Updated frontend to 20190319.0

balloob committed:
Bumped version to 0.90.0b6

balloob committed:
Updated frontend to 20190319.1

balloob committed:
Bumped version to 0.90.0b7

balloob committed:
Updated frontend to 20190320.0

Swamp-Ig committed:
Fixed typing errors (#22207)

balloob committed:
Bumped version to 0.90.0

balloob committed:
Merge pull request #22216 from home-assistant/rc

0.90.0

Adminiuga committed:
Fix ZHA force polled entities. (#22222)

## Description:
Fix "force_polled" ZHA entities.

## Checklist:
  - [x] The code change is tested and works locally.
  - [x] Local tests pass with `tox`. **Your PR cannot be merged unless tests pass**
  - [x] There is no commented out code in this PR.

[ex-requir]: https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/components/keyboard/__init__.py#L14
[ex-import]: https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/components/keyboard/__init__.py#L23

hunterjm committed:
Stream fixes (#22238)

* fix issues with out of order packets, and empty first packet on some IP camera models

* do not skip the first packet

michaelarnauts committed:
Allow on/off on tado climate component. (#22242)

* Allow on/off on tado climate component. Bump python-tado version. Patch from @wmalgadey

* Revert wrongly change in tado device tracker

andrewsayre committed:
Fix validate webhook requirements (#22248)

kkr16 committed:
Bump total-connect-client to 0.25, fixing issue with Total Connect (#22230)

* Bump total-connect-client to 0.25

* Bump version in requirements_all.txt

pvizeli committed:
Update Hass-NabuCasa 0.9 (#22258)

balloob committed:
Updated frontend to 20190321.0

balloob committed:
Bumped version to 0.90.1

balloob committed:
Lint

pvizeli committed:
Update hass-nabucasa 0.10 (#22267)

balloob committed:
Merge pull request #22265 from home-assistant/rc

0.90.1

ktnrg45 committed:
Fix ps4 no creds with additional device (#22300)

* Fix no creds with additional device.

* Update config_flow.py

hunterjm committed:
Prefer TCP for RTSP streams (#22338)

## Description:

For RTSP streams, set the `prefer_tcp` FFMPEG flag.  This should resolve some of the "green feed" issues that some users are reporting, likely due to packets being lost over UDP on their network.

Resources: [FFMPEG protocols documentation](https://ffmpeg.org/ffmpeg-protocols.html#rtsp)

## Checklist:
  - [x] The code change is tested and works locally.
  - [x] Local tests pass with `tox`. **Your PR cannot be merged unless tests pass**
  - [x] There is no commented out code in this PR.

pvizeli committed:
Update hass-nabucasa & fix state (#22385)

* Update hass-nabucasa & fix state

* Fix lint

hunterjm committed:
reset unsub to None on timeout (#22404)

pvizeli committed:
Enable hass.io panel without ping (#22388)

* Enable hass.io panel without ping

* fix tests

balloob committed:
Bumped version to 0.90.2

Commit:
Add CasaTunes support

Commit:
Add CasaTunes support

fbradyirl committed:
ciscomobilityexpress pypi version update (#22431)

* Bump pip

* Bump ciscomobilityexpress to 0.1.4

balloob committed:
Merge pull request #22420 from home-assistant/rc

0.90.2

pvizeli committed:
Update homeassistant-pyozw to 0.1.3 (#22433)

fbradyirl committed:
Adding conf for deep standby, wake and specific source bouquet of Enigma2 (#22393)

* - adding deep standby, conf for wake and specify source bouquet

* set defaults to strings

* bump pip

* bump pip

* bump pip

* bump pip

* bump pip

* bump pip

balloob committed:
Fix test name (#22421)

thomasloven committed:
Lower severity level of log messages from http.view (#21091)

pvizeli committed:
Update ha-ffmpeg 2.0 (#22427)

zewelor committed:
Add myself as codeowner for yeelight component (#22438)

Adminiuga committed:
Update ZHA component CODEOWNERS (#22452)

zewelor committed:
Add support for yeelight ceiling ambilight (#22346)

Swamp-Ig committed:
Bootstrap to start registry loading early (#22321)

* Registries store directly in data on loading.

* Loading registries concurent with stage 1.

* Removed comments

uchagani commented on homeassistant/components/casatunes/media_player.py:
```diff
@@ -0,0 +1,270 @@
+    for zone in zones:
+        try:
+            zone_devices.append(CasaTunesZone(zone))
+        except TypeError:
```
Under what conditions would this error be thrown?  Do we want the platform to retry if the initial attempt doesn't work?

resoai replied:
There could (hypothetically) be a network issue or CasaTunes could be updating.

uchagani replied:
In that case you may want to throw a PlatformNotReady error that way the Home Assistant system can retry in a few seconds to try and setup the platform.  Without that, a user will need to restart home assistant to try getting this platform to work.

resoai replied:
Let's wait for Travis to run his tests again

Commit:
Added PlatformNotReady

Commit:
Added PlatformNotReady

Kane610 committed:
Axis devices support device registry (#22367)

* Add support for device registry

* Fix test

ljmerza committed:
Use voluptuous error string for websocket validation error (#21883)

* use voluptuous error string to websocket validation error

* added exception logging to websocket error

* add detailed message to websocket validation error

* add error message to websocket validation error

* Add humanize error for websocket invalid vol error

* Add humanize error for websocket invalid vol error

* Add humanize error for websocket invalid vol error

rtclauss committed:
Add hourly forecasts to Dark Sky (#21820)

fredrike committed:
Add switches to control Daikin Airbase zones (#22417)

* initial AirBase zone support

* fix zone name

* version bump pydaikin

* don't use get()

heythisisnate committed:
better algorithm for computing unique_id (#22389)

GidoHakvoort committed:
Solaredge new sensors (#21047)

* Remove unused hass parameter for SolarEdgeData

* Add factory to create different types of sensors
Rename SolarEdgeSensor to SolarEdgeOverviewSensor
Rename SolarEdgeData to SolarEdgeOverviewDataService
Remove unused hass parameter in SolarEdgeOverviewDataService

* Add SolarEdgeDetailsDataService to retrieve details data
Add SolarEdgeDetailsSensor to report details data
Add abstract class SolarEdgeSensor
Add details sensor types

* Combine multiple details sensor into one sensor with attributes

* Fix pylint and flake8 errors

* Resolve conflict with solaredge component update

* Add SolarEdgeInventoryDataService to retrieve inventory information
Add SolarEdgeInventorySensor to view inventory information
Add inverters to monitored_conditions

* Fix pylint and flake8 errors

* Add additional monitored variables for solaredge

* Add new sensors to solaredge component

* Add SolarEdgePowerFlowDataService
Add SolarEdgePowerFlowSensor
Add new monitored_conditions for power consumption and grid, load and
solar power production/consumption

* Set entity_id for each sensor based on platform and sensor type

* Fix flake8 and pylint errors

* Add check for connections in return data

* Fix pylint and flake8 errors

* Renamed state_attributes to device_state_attributes
Moved request import to top

* Remove explicit definition of entity_id

* Fix pylint and flake8 errors

* Add check for None before adding sensor

* Update SolarEdgeSensorFactory with initial dict which maps sensor_key to
entity class and data service

* Update attribute values to snakecase
Added stingcase as requirement

* Update requirements_all.txt to include stringcase for solaredge

* Update some initial values for data and unit_of_measurement
Update sensor factory

awarecan committed:
Fix aws.notify platform schema (#22374)

* Fix aws component notify platform schema

* Address code review comment

* Do not allow load aws.notify from notify component

* Revert unrelated translation update

* Review comment

amelchio committed:
Do data extraction in sensors (#22444)

* Do data extraction in sensors

* Hopefully fix lint

Jc2k committed:
Remove homekit_controller duplicate legacy pairing loader code (#22442)

renemarc committed:
Cache GTFS metadata and expose utility attributes (breaking change) (#20966)

## Description:
Current sensor updates run 7 additional SQLite database queries to populate attributes, on top of the bus schedule queries themselves. Double that if you have two sensors. That leads to a lot of slowdowns for everything else when using an SD card!

Considering that some data never changes (agency, routes...) and that others like departure times are good until invalidated, let's fetch such metadata at first then only when relevant changes do occur.

**Breaking Change:**
GTFS sensor attributes are now named using the standard snake_case format.

### Work performed:
- All metadata queries are now cached.
- Metadata queries are now all regrouped in the `update()` method.
- Attributes assembling is now done in ~~`device_state_attributes()` where it belongs.~~ in a utility method called from `update()`, for code clarity and since there is potential I/O from SQLAlchemy.
- As a bonus, many metadata entries with cryptic values have complementary entries added that provide easier to use data:
	- .\* Stop Drop Off Type: .\* Stop Drop Off Type **State** -> (string, unknown)
	- .\* Stop Pickup Type: .\* Stop Pickup Type **State** -> (string, unknown)
	- .\* Stop Timepoint: .\* Stop Timepoint **Exact** -> boolean
	- .\* Station Location Type: .\* Station Location Type **Name** -> string
	- .\* Wheelchair Boarding: .\* Wheelchair Boarding **Available** -> (boolean, unknown)
	- Route Type: Route Type **Name** (string)
	- Trip Bikes Allowed: Trip Bikes Allowed **State** -> (boolean, unknown)
	- Trip Wheelchair Access: Trip Wheelchair Access **Available** -> (boolean, unknown)
- Attribute names are now using snake_case.
- Added type hints.

**Related issue (if applicable):** fixes #21222

## Checklist:
  - [x] The code change is tested and works locally.
  - [x] Local tests pass with `tox`. **Your PR cannot be merged unless tests pass**
  - [x] There is no commented out code in this PR.

dilruacs committed:
Modify check for ADB public key (#22378)

* Remove check for public key

* Remove has_adb_files, directly call cv.isfile

* Check for missing adbkey.pub, create dummy if not found

* Reorder imports

* Bumped androidtv library version, deactivated pubkey test

* Code works without pubkey, removed function

* Removed "import os", not needed anymore

* Bump library version

rohankapoorcom committed:
Centralize all usages of `host` as a configuration param to the constant CONF_HOST (#22458)

* Centralize all usages of  as a configuration param to the constant CONF_HOST

* Clean up test

andrewsayre committed:
Don't return cover position when not supported (#22484)

Adminiuga committed:
Targeted ZHA permit joins. (#22482)

* Targeted ZHA permit service.

* Convert IEEE string to EUI64 usiv vol schema.

* Update test units.

* Lint.

isort imports.

syssi committed:
Fix auto discovery of yeelights (#22481)

* Fix auto discovery of yeelights

* Fix lint

heythisisnate committed:
fix inverse state changes for binary sensors (#22479)

Kane610 committed:
Fix events so they work with multiple devices (#22477)

Danielhiversen committed:
Improve Sensibo error handling (#22475)

* Handle sensibo exception

* improve sensibo

bieniu committed:
Add new data fields and bump python-join-api (#22472)

* Add new data fields and bump python-join-api

* Update __init__.py

fbradyirl committed:
Bump pypi again for Cisco Mobility Express (#22467)

* Fix empty device clId

Fix for https://github.com/fbradyirl/ciscomobilityexpress/issues/13

* Update requirements_all.txt

jackwilsdon committed:
Return percentage information in Alexa Smart Home response (#22440)

Jc2k committed:
Move HKDevice into connection (#22430)

amelchio committed:
Incoming SMS handling for netgear_lte (#22402)

* Fire netgear_lte events for incoming SMS

* Add netgear_lte.delete_sms service call

* Fix log statement

* Add services.yaml

hfurubotten committed:
Entur upgrade to v0.2.0: async polling, number of departures, omit non boarding departures (#22001)

* The Great Migration

* removed outdated tests

* Upgraded enturclient to v0.2.0
- client has gone async
- configurable if non boarding departures should be omitted from result
- configurable how many departures should be shown
- more clear which attributes are included and when

* reduced nesting and clearing of attributes

* Removed test data

* ensure attribution and stop id in attributes

* docstring fixes

* fix rebase errors
went a bit too fast on continue on one of the conflicts..

* fix test requirements comment from gen

rohankapoorcom committed:
Mopar split (#21526)

* Split out mopar into a component and sensor platform

* Add the mopar switch platform

* Add the mopar lock platform

* Clean up and bump version

* Update per review

* Re-add service to trigger horn

* Clean up again

* Don't call async from sync context

* Lint

* Implement changes from review

* Lint

* A little more clean up

bachya committed:
Fix too-abrubt SimpliSafe data refresh termination on error (#22466)

balloob committed:
Fix syntax error

robbiet480 committed:
Great migration notify (#22406)

* Move notify platforms into components

* Move notify tests

* Fix notify tests

* More fixes

* Update requirements

* Update .coveragerc

* Run gen reqs

balloob committed:
Print error instead of warning for custom platforms in legacy format (#22486)

* Legacy platform format prints error

* Enforce no partial overlays

cliffordwhansen committed:
Add game and app media types (#22459)

* Added game and app media types

* Changed media type to game from music

* Removed app type as it is not used yet

balloob committed:
Updated frontend to 20190327.0

balloob committed:
Update translations

hunterjm committed:
Stream Record Service (#22456)

* Initial commit of record service for live streams

* fix lint

* update service descriptions

* add tests

* fix lint

awarecan committed:
Add trusted_users in trusted networks auth provider (#22478)

balloob committed:
Merge remote-tracking branch 'origin/master' into dev

balloob committed:
Version bump to 0.92.0dev0

balloob committed:
Fix YAML

awarecan committed:
Fix dev branch (#22493)

resoai commented:
@uchagani what shall we do now?

awarecan committed:
A very basic Circleci setup (#22503)

* Add circleci support

* Add buildpack-deps

* Install libudev-dev

* sudo

* always run test

* Add test report

* no sugar

* quite pytest

* better junit test result

* Add $CODE_COVERAGE env var

awarecan committed:
Fix Circleci config (#22509)

* Add libav depends on circleci

* tweak circleci config

awarecan committed:
Remove botocore dependency from credstash script (#22511)

* Remove botocore dependency from credstash script

* Update requirements_all.txt

* Update pylintrc

* Update credstash.py

alengwenus committed:
Add LCN binary_sensor component (#22341)

awarecan committed:
Fix lint on dev (#22512)

## Description:

Fix a lint issue in credstash script.


**Related issue (if applicable):** fixes #<home-assistant issue number goes here>

**Pull request in [home-assistant.io](https://github.com/home-assistant/home-assistant.io) with documentation (if applicable):** home-assistant/home-assistant.io#<home-assistant.io PR number goes here>

## Example entry for `configuration.yaml` (if applicable):
```yaml

```

## Checklist:
  - [ ] The code change is tested and works locally.
  - [ ] Local tests pass with `tox`. **Your PR cannot be merged unless tests pass**
  - [ ] There is no commented out code in this PR.

If user exposed functionality or configuration variables are added/changed:
  - [ ] Documentation added/updated in [home-assistant.io](https://github.com/home-assistant/home-assistant.io)

If the code communicates with devices, web services, or third-party tools:
  - [ ] New dependencies have been added to the `REQUIREMENTS` variable ([example][ex-requir]).
  - [ ] New dependencies are only imported inside functions that use them ([example][ex-import]).
  - [ ] New or updated dependencies have been added to `requirements_all.txt` by running `script/gen_requirements_all.py`.
  - [ ] New files were added to `.coveragerc`.

If the code does not interact with devices:
  - [ ] Tests have been added to verify that the new code works.

[ex-requir]: https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/components/keyboard/__init__.py#L14
[ex-import]: https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/components/keyboard/__init__.py#L23

easink committed:
Add HEOS media player component (#21721)

## Description:
Denon HEOS media player.

**Pull request in [home-assistant.io](https://github.com/home-assistant/home-assistant.io) with documentation (if applicable):** home-assistant/home-assistant.io#8848

## Example entry for `configuration.yaml` (if applicable):
```yaml
heos:
  host: HEOS-1                              
```

## Checklist:
  - [X] The code change is tested and works locally.
  - [X] Local tests pass with `tox`. **Your PR cannot be merged unless tests pass**
  - [X] There is no commented out code in this PR.

If user exposed functionality or configuration variables are added/changed:
  - [X] Documentation added/updated in [home-assistant.io](https://github.com/home-assistant/home-assistant.io)

If the code communicates with devices, web services, or third-party tools:
  - [X] New dependencies have been added to the `REQUIREMENTS` variable ([example][ex-requir]).
  - [X] New dependencies are only imported inside functions that use them ([example][ex-import]).
  - [X] New or updated dependencies have been added to `requirements_all.txt` by running `script/gen_requirements_all.py`.
  - [X] New files were added to `.coveragerc`.

If the code does not interact with devices:
  - [ ] Tests have been added to verify that the new code works.

[ex-requir]: https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/components/keyboard/__init__.py#L14
[ex-import]: https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/components/keyboard/__init__.py#L23


Co-authored-by: Andrew Sayre <6730289+andrewsayre@users.noreply.github.com>

yosilevy committed:
Add google calendar max_results config option (#21874)

* Added max_results config capability to google calendar (people are creating custom components just to override that)

* Dummy commit

* Dummy commit 2

* Changed to positive_int

* Removed double imports

andrewsayre committed:
Fix .coveragerc from merge/rebase (#22516)

* Fix coveragerc

* Fix coveragerc

mvn23 committed:
Update pyotgw to 0.4b3 (#22496)

bachya committed:
Speed up status updating in SimpliSafe (#22506)

* Speed up status updating in SimpliSafe

* Linting

* Member comments

Commit:
Bumped version of casatunes

Commit:
Bumped version of casatunes

Kane610 committed:
Axis component reflect device availability (#22401)

aav7fl committed:
throw `PlatformNotReady` if unable to connect (#22515)

Throw `PlatformNotReady` for when the device disconnects, or when the Home Assistant is booting and the ADB server is not ready yet.

Commit:
Merge branch 'dev' of https://github.com/resoai/home-assistant into dev

scyto commented:
@resoai is there a reason this was closed?  i was really hoping to use this?

scyto commented:
for anyone else arriving here it can be installed as a custom component by copying @resoai 's files from https://github.com/resoai/HomeAssistantConfig/tree/master/custom_components/casatunes :-)

resoai commented:
I have become a father and could not find time to finalise it. Skipping and track progress are mostly broken.
